### PR TITLE
[MM-43336] Fix some e2e tests

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Fix e2e tests that broke after.

A lot of them depended on a removed modal, and the features have not all been ported.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43336

#### Screenshots
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by pvev — APPROVED** [review]
Great, thanks @JulienTant :shipit:

File: e2e/cypress/tests/integration/messaging/center_channel_rhs_overlap_spec.js [hunk 1]
```diff
@@ -805,41 +805,6 @@ describe('Messaging', () => {
             cy.get('#edit_textbox', {timeout: TIMEOUTS.FIVE_SEC}).type('{enter}');
         });
     });
-
-    it('MM-T2227 Channel shortlinking - can edit', () => {
```
Comment (julmondragon) [inline e2e/cypress/tests/integration/messaging/center_channel_rhs_overlap_spec.js:809]: Was this feature not replaced/moved?

File: e2e/cypress/tests/integration/accessibility/accessibility_dropdowns_spec.js [hunk 2]
```diff
@@ -75,8 +75,6 @@ describe('Verify Accessibility Support in Dropdown Menus', () => {
         menuItems.forEach((item) => {
             // * Verify that the menu item is focused
             if (item === 'View Info' || item === 'Mute Channel' || item === 'Leave Channel') {
-                cy.uiGetChannelMenu().findByText(item).parent().parent().should('be.focused');
```
Comment (Phrynobatrachus) [inline e2e/cypress/tests/integration/accessibility/accessibility_dropdowns_spec.js:78]: thanks for looking at this, I wasn't sure about what changes may have caused issues here. I think the `if/else` can be removed entirely, so only line 78 is left in the `forEach`. I checked that a moment ago and it appears to assert on each menu item correctly.

File: e2e/cypress/tests/integration/messaging/center_channel_rhs_overlap_spec.js (hunk 1 above)
Comment (JulienTant) [reply e2e/cypress/tests/integration/messaging/center_channel_rhs_overlap_spec.js:809]: The link has been completely removed from the RHS

File: e2e/cypress/tests/integration/accessibility/accessibility_dropdowns_spec.js (hunk 2 above)
Comment (JulienTant) [reply e2e/cypress/tests/integration/accessibility/accessibility_dropdowns_spec.js:78]: Oh that was my intent... obviously not what I did tho haha! Thanks

Comment (JulienTant) [reply e2e/cypress/tests/integration/accessibility/accessibility_dropdowns_spec.js:78]: Oh that was my intent... obviously not what I did tho haha! Thanks

**Review by Phrynobatrachus — APPROVED** [review]
The E2E run above looks fine, all tests touched here passed. Other failures are unrelated internal issues. Local double checks also LGTM.

https://automation-dashboard.vercel.app/cycles/010f970c-6004-4df0-92ea-0e21638bf804

https://git.internal.mattermost.com/qa/cypress-ui-automation/-/jobs/772115
(internal failure link, appears to be registry issues again)

Going to take Steve off the hook and merge. Thanks for the fixes!

**Pull request merged**